pkg/batchd/policy/drf/drf.go

```diff
@@ -98,6 +99,27 @@ func (drf *drfScheduler) Allocate(queues []*cache.QueueInfo, nodes []*cache.Node
 				break
 			}
 		}
+
+		// If policy could not assign minAvailable for a PodSet, then recycle assigned resources
+		if !psi.meetMinAvailable() {
+			assignedPending := psi.resetAndGetAssignedPod()
```

Comment:
I think we need to trace what we had assign in this loop instead of reset all assigned pods.

Reply:
got it